pkg/translate/translateValue.go

```diff
@@ -0,0 +1,554 @@
+				"{{.ValueComponentName}}.priorityClassName":   {"{{.FeatureName}}.Components.{{.ComponentName}}.Common.K8s.PriorityClassName", nil},
+				"{{.ValueComponentName}}.readinessProbe":      {"{{.FeatureName}}.Components.{{.ComponentName}}.Common.K8s.ReadinessProbe", nil},
+				"{{.ValueComponentName}}.replicaCount":        {"{{.FeatureName}}.Components.{{.ComponentName}}.Common.K8s.ReplicaCount", nil},
+				"{{.ValueComponentName}}.resources":           {"{{.FeatureName}}.Components.{{.ComponentName}}.Common.K8s.Resources", nil},
+			},
+			ValuesToComponentName: map[string]name.ComponentName{
+				"pilot":                         name.PilotComponentName,
+				"galley":                        name.GalleyComponentName,
+				"sidecarInjectorWebhook":        name.SidecarInjectorComponentName,
+				"mixer.policy":                  name.PolicyComponentName,
+				"mixer.telemetry":               name.TelemetryComponentName,
+				"security":                      name.CitadelComponentName,
+				"nodeagent":                     name.NodeAgentComponentName,
+				"certmanager":                   name.CertManagerComponentName,
+				"gateways.istio-ingressgateway": name.IngressComponentName,
+				"gateways.istio-egressgateway":  name.EgressComponentName,
+			},
+			NamespaceMapping: map[string][]string{
+				"global.istioNamespace":     {"security.components.namespace"},
+				"global.telemetryNamespace": {"telemetry.components.namespace"},
+				"global.policyNamespace":    {"policy.components.namespace"},
+				"global.configNamespace":    {"configManagement.components.namespace"},
+			},
+		},
+	}
+	// Component enablement mapping. Ex "{{.ValueComponent}}.enabled": {"{{.FeatureName}}.Components.{{.ComponentName}}.Common.enabled}", nil},
+	// Feature enablement mapping. Ex: "{{.ValueComponent}}.enabled": {"{{.FeatureName}}.enabled}", nil},
+	componentEnablementPattern = "{{.FeatureName}}.Components.{{.ComponentName}}.Common.Enabled"
+	componentValuesPattern     = "{{.FeatureName}}.Components.{{.ComponentName}}.Common.Values"
+)
+
+// initAPIMapping generate the reverse mapping from original translator apiMapping
+func (t *ValueYAMLTranslator) initAPIMapping(vs version.MinorVersion) error {
+	ts, err := NewTranslator(vs)
+	if err != nil {
+		return err
+	}
+	for valKey, outVal := range ts.APIMapping {
+		t.APIMapping[outVal.outPath] = &Translation{valKey, nil}
+	}
+	return nil
+}
+
+// initK8SMapping generates the k8s settings mapping for components that are enabled based on templates
+func (t *ValueYAMLTranslator) initK8SMapping(valueTree map[string]interface{}) {
+	outputMapping := make(map[string]*Translation)
+	for valKey, componentName := range t.ValuesToComponentName {
+		featureName := name.ComponentNameToFeatureName[componentName]
+		if featureName == "" {
+			continue
+		}
+		cnEnabled, err := name.IsComponentEnabledFromValue(valKey, valueTree)
+		if err != nil {
+			log.Errorf("Error checking component:%s enablement, skip k8s mapping", componentName)
```

Comment:
I'd think this error is serious enough you'd want to return it and fail the translator initialization.

Reply:
done